src/base_convert.rs

```diff
@@ -124,36 +101,42 @@ impl<const BITS: usize, const LIMBS: usize> Uint<BITS, LIMBS> {
         if base < 2 {
             return Err(BaseConvertError::InvalidBase(base));
         }
-
-        let mut tail = digits.into_iter();
-        match tail.next() {
-            Some(digit) => Self::from_base_le_recurse(digit, base, &mut tail),
-            None => Ok(Self::ZERO),
+        if BITS == 0 {
+            for digit in digits {
+                if digit >= base {
+                    return Err(BaseConvertError::InvalidDigit(digit, base));
+                }
+                if digit != 0 {
+                    return Err(BaseConvertError::Overflow);
+                }
+            }
+            return Ok(Self::ZERO);
         }
-    }
 
-    /// This is the recursive part of [`Uint::from_base_le`].
-    ///
-    /// We drain the iterator via the recursive calls, and then perform the
-    /// same construction loop as [`Uint::from_base_be`] while exiting the
-    /// recursive callstack.
-    #[inline]
-    fn from_base_le_recurse<I: Iterator<Item = u64>>(
-        digit: u64,
-        base: u64,
-        tail: &mut I,
-    ) -> Result<Self, BaseConvertError> {
-        if digit > base {
-            return Err(BaseConvertError::InvalidDigit(digit, base));
+        let mut iter = digits.into_iter();
+        let mut result = Self::ZERO;
+        let mut power = Self::from(1);
+        while let Some(digit) = iter.next() {
+            if digit >= base {
+                return Err(BaseConvertError::InvalidDigit(digit, base));
+            }
+            let term = power.checked_mul(Self::from(digit)).ok_or(BaseConvertError::Overflow)?;
+            result = result.checked_add(term).ok_or(BaseConvertError::Overflow)?;
+            if let Some(next_power) = power.checked_mul(Self::from(base)) {
```

Comment:
These three lines can be combined into a simultaneous iteration over both `power` and `result`. It will similar to, but a bit more complex than the loop in `from_base_be`. As it is currently it doesn't make good use of multiplicants being single limbs.